rio/src/RIO/Process.hs

```diff
@@ -258,6 +268,10 @@ exeSearchPathL = processContextL.to pcPath
 
 -- | Create a new 'ProcessContext' from the given environment variable map.
 --
+-- On Windows, the filename extensions associated with executables are from the
```

Comment:
Again, this should be related to an identifier which allows these to be overridden.

Reply:
I have made this much shorter and added an 'only' to 'On Windows only, ...' to reinforce that non-Windows users can skip it. As there is no exported identifier, I have kept the documentation with this most general of the 'smart' constructors - and avoided the repetition with the more specialised 'smart' constructor `mkDefaultProcessContext` by referring back to `mkProcessContext`. I think some Haddock documentation is necessary because behaving as if the system `PATHEXT` variable is defined with a default value, when it is not present in the environment, is not something you might anticipate.